packages/vaadin-virtual-list/src/iron-list.js

```diff
@@ -164,7 +166,8 @@ export const PolymerIronList = Class({
    * The height of the physical content that isn't on the screen.
    */
   get _hiddenContentSize() {
-    return this._physicalSize - this._viewportHeight;
+    var size = this.grid ? this._physicalRows * this._rowHeight : this._physicalSize;
```

Comment:
As mentioned [here](https://github.com/vaadin/web-components/pull/321#discussion_r626628993), this file is a direct copy-paste from the [iron-list source code](https://github.com/PolymerElements/iron-list/blob/master/iron-list.js#L396-L1983).

In scope of this PR, I updated the source to newest version so there are some changes. The iron-list version used in grid's fork was a bit outdated and had some bugs that have since been fixed.

I don't consider cleaning up iron-list source code (other than removal of some unnecessary code) as something that should be done in scope of this PR. This was also discussed [here](https://github.com/vaadin/web-components/pull/321#discussion_r627404302)